### ISSUE TITLE
HAML SASS Directive fails when passed an 8-digit hex value.

### ISSUE THREAD

**Issue by dlinch** [haml/haml#997]
tl;dr: The HAML Sass engine fails to parse 8-digit hex codes as valid css.

We are working on a project that allows theming, and we're injecting a base color into the application head tag to build our theme out from.

Right now it looks like:

```
:sass
  $primary-color: #{@color}
  $secondary-color: #{@second_color}
  # Do stuff with the colors
```

Our primary color comes through as a normal hex value "#000000" or "#000", but our secondary is just our primary color that we programmatically add an alpha value onto. When we go to load the page, the HAML engine errors and the error message points to our 8-digit hex value as invalid CSS. 
4-digit and 8-digit hex codes have pretty wide adoption at the moment: [https://caniuse.com/#search=hex](https://caniuse.com/#search=hex)
So I would argue that the Sass Engine in HAML should support it.

I found a closed issue from 8 years ago that might point a contributor in the right direction #168

**Comment by k0kubun** [comment]
Haml delegates Sass handling to Tilt, so this issue should go to https://github.com/rtomayko/tilt first.
